src/util/RegistrationFormFactory.js

```diff
@@ -17,6 +18,72 @@ define([
 ], function (Okta, SchemaFormFactory, TextBox) {
 
   var _ = Okta._;
+  
+  var checkSubSchema = function(subSchema, value, model) {
+    var minLength = subSchema.get('minLength');
+    var maxLength = subSchema.get('maxLength');
+    var regex = subSchema.get('format');
+    
+    if (_.isNumber(minLength)) {
+      if (value.length < minLength) {
+        return false;
+      }
+    }
+    
+    if (_.isNumber(maxLength)) {
+      if (value.length > maxLength) {
+        return false;
+      }
+    }
+    
+    if (_.isString(regex)) {
+
+      if (regex === '^[#/userName]') {
+        var username = model.get('userName').toLowerCase();
+        var fieldVal = value.toLowerCase();
+        if (username && fieldVal.indexOf(username) !== -1) {
+          return false;
+        }
+      } else {
+        if (!new RegExp(regex).test(value)) {
+          return false;
+        }
+      }
+    }
+    
+    return true;
+  };
+  
+  
+  var checkSubSchemas = function (fieldName, model, subSchemas, showError) {
+    var value = model.get(fieldName);
+    if (!_.isString(value)) {
+      return;
+    }
+
+    subSchemas.each(function(subSchema, index) {
+      var ele = Okta.$('#subschemas-' + fieldName + ' .subschema-' + index);
+      //hide password complexity if no password
+      if (value) {
+        ele.children('p').removeClass('default-schema');
+      } else {
+        ele.children('p').addClass('default-schema');
+      }
+      ele.removeClass('subschema-satisfied subschema-unsatisfied subschema-error');
+      if (checkSubSchema(subSchema, value, model)) {
+        ele.addClass('subschema-satisfied');
+        ele.find('p span').removeClass('error error-16-small');
+        ele.find('p span').addClass('confirm-16');
+      } else {
+        if (showError) {
+          ele.find('p span').removeClass('confirm-16');
+          ele.find('p span').addClass('error error-16-small');
+          ele.addClass('subschema-error');
+          ele.addClass('subschema-unsatisfied');
```

Comment:
Nit: Can combine those to ``ele.addClass('subschema-error subschema-unsatisfied');``